File: Activities/SettingsActivity.cs
using Android.App;
using Android.OS;
using AndroidX.AppCompat.App;
using AndroidX.Preference;
using ScreenLockApp.Themes;

namespace ScreenLockApp.Activities;

[Activity(Label = "@string/settings_title")]
public class SettingsActivity : AppCompatActivity
{
    protected override void OnCreate(Bundle? savedInstanceState)
    {
        // Apply theme before setContentView
        ThemeManager.ApplyTheme(this);
        
        base.OnCreate(savedInstanceState);
        SetContentView(Resource.Layout.activity_settings);
        
        // Load preferences fragment
        if (savedInstanceState == null)
        {
            SupportFragmentManager
                .BeginTransaction()
                .Replace(Resource.Id.settings_container, new SettingsFragment())
                .Commit();
        }
        
        // Enable back button
        SupportActionBar?.SetDisplayHomeAsUpEnabled(true);
    }
    
    public override bool OnSupportNavigateUp()
    {
        Finish();
        return true;
    }
}

public class SettingsFragment : PreferenceFragmentCompat
{
    public override void OnCreatePreferences(Bundle? savedInstanceState, string? rootKey)
    {
        SetPreferencesFromResource(Resource.Xml.preferences, rootKey);
        
        // Set up strict mode listener
        var strictMode = FindPreference("strict_mode");
        if (strictMode != null)
        {
            strictMode.PreferenceChange += StrictMode_PreferenceChange;
        }
    }
    
    private void StrictMode_PreferenceChange(object? sender, Preference.PreferenceChangeEventArgs e)
    {
        if (e.NewValue is bool strictEnabled && strictEnabled)
        {
            // Override tolerances to 0
            var prefs = PreferenceManager.GetDefaultSharedPreferences(RequireContext());
            prefs?.Edit()
                ?.PutInt("tolerance_minutes", 0)
                ?.PutInt("battery_tolerance", 0)
                ?.Apply();
        }
    }
}

File: app/src/main/java/com/nkkinsoft/screenlockapp/SettingsActivity.java
package com.nkkinsoft.screenlockapp;

import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.SwitchPreferenceCompat;

/**
 * Settings activity for configuring app preferences including theme settings.
 */
public class SettingsActivity extends AppCompatActivity {
    private ThemeManager themeManager;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // Apply theme before setContentView
        themeManager = new ThemeManager(this);
        themeManager.apply();
        
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        
        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.settings_container, new SettingsFragment())
                    .commit();
        }
        
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
    }
    
    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
    
    public static class SettingsFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.preferences, rootKey);
            
            // Handle high contrast toggle
            SwitchPreferenceCompat highContrastPref = findPreference("high_contrast");
            if (highContrastPref != null) {
                highContrastPref.setOnPreferenceChangeListener((preference, newValue) -> {
                    // Recreate activity to apply new theme
                    if (getActivity() != null) {
                        getActivity().recreate();
                    }
                    return true;
                });
            }
        }
    }
}
